datastore/core/endpoints/Datastore.stream.ts

```diff
@@ -75,3 +50,88 @@ export default new DatastoreApiHandler('Datastore.stream', {
     };
   },
 });
+
+async function extractFunctionOutputs(
+  manifestWithStats: IDatastoreManifestWithStats, 
+  datastore: Datastore, 
+  request: IDatastoreApis['Datastore.stream']['args'],
+  context: IDatastoreApiContext, 
+  paymentProcessor: PaymentProcessor,
+): Promise<any[]> {
+  await paymentProcessor.createHold(
+    manifestWithStats,
+    [{ functionName: request.name, id: 1 }],
+    request.pricingPreferences,
+  );
+
+  validateFunctionCoreVersions(manifestWithStats, request.name, context);
+
+  return await context.workTracker.trackRun(
+    (async () => {
+      const options: IFunctionExecOptions<any> = {
+        input: request.input,
+        authentication: request.authentication,
+        affiliateId: request.affiliateId,
+        payment: request.payment,
+      };
+
+      for (const plugin of Object.values(DatastoreCore.pluginCoresByName)) {
+        if (plugin.beforeExecFunction) await plugin.beforeExecFunction(options);
+      }
+
+      const results = datastore.functions[request.name].runInternal(options);
+      for await (const result of results) {
+        context.connectionToClient.sendEvent({
+          listenerId: request.streamId,
+          data: result,
+          eventType: 'FunctionStream.output',
+        });
+      }
+      return results;
+    })(),
+  );
+}
+
+function extractTableOutputs(
+  datastore: Datastore, 
+  request: IDatastoreApis['Datastore.stream']['args'],
+  context: IDatastoreApiContext, 
+): any[] {
+  let storage: DatastoreStorage;
+  if (request.versionHash) {
+    const storagePath = context.datastoreRegistry.getStoragePath(request.versionHash);
+    storage = new DatastoreStorage(storagePath);
+  } else {
+    context.connectionToClient.datastoreStorage ??= new DatastoreStorage();
+    storage = context.connectionToClient?.datastoreStorage;
+  }
+
+  const db = storage.db;
+  const schema = datastore.tables[request.name].schema;
+
+  const fields = ['*'];
+  const where: string[] = [];
+  const boundValues: string[] = [];
+
+  for (const field of Object.keys(request.input || {})) {
+    const value = request.input[field];
```

Comment:
Seems like a SQL injection attack vector here without checking these fields.

Reply:
Can we chat about this?